datasets/plugins/batch/batch_base_plugin.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import pandas as pd
+
+from datasets import Mode
+from datasets._typing import ColumnNames
+from datasets.dataset_plugin import DatasetPlugin
+from datasets.exceptions import InvalidOperationException
+
+
+_logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    import dask.dataframe as dd
+    from pyspark.sql import DataFrame as SparkDataFrame
+
+
+class BatchBasePlugin(DatasetPlugin, dict):
+    """
+    The base plugin for the BATCH execution context.
+    """
+
+    _dataset_path_func: Callable = None
+
+    def __init__(
+        self,
+        name: str,
+        hive_table_name: str,
+        logical_key: str = None,
+        columns: Optional[ColumnNames] = None,
+        run_id: Optional[str] = None,
+        mode: Mode = Mode.READ,
+        partition_by: Optional[ColumnNames] = None,
+    ):
+        self.hive_table_name = hive_table_name
+        self.partition_by = partition_by
+        self.program_name = self._executor.current_program_name
+        super(BatchBasePlugin, self).__init__(
+            name=name,
+            logical_key=logical_key,
+            columns=columns,
+            run_id=run_id,
+            mode=mode,
+        )
+
+        def set_name(key: str, value: Optional[object]):
+            if value:
+                self[key] = value
+
+        dict.__init__(self, name=name, hive_table_name=hive_table_name, mode=self.mode.name)
+        set_name("logical_key", logical_key)
+        set_name("columns", columns)
+        set_name("run_id", run_id)
+        set_name("partition_by", partition_by)
+
+    def _get_filters_columns(
+        self,
+        columns: Optional[ColumnNames] = None,
+        run_id: Optional[str] = None,
+        partitions: Optional[dict] = None,
+    ) -> Tuple[Optional[list], Optional[Iterable[str]]]:
+        read_columns = self._get_read_columns(columns)
+        filters: Optional[List[Tuple]] = None
+        query_run_id = run_id if run_id else self.run_id
+        if query_run_id:
+            filters = [("run_id", "=", query_run_id)]
+
+        if partitions:
+            if filters is None:
+                filters = []
+            for key, value in partitions.items():
+                filters.append((key, "=", value))
```

Comment:
would there be cases where we support > or < here? Don't know much about the pandas implementation of filters in read_parquet and whether it supports this or not..